hermes/probe/delete_file.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Delete_file implements the probe operation for deleting a file in a
+// storage system.
+// Package probe implements the probe that Hermes uses to monitor
+// a storage system.
+
+package probe
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes"
+	"google.golang.org/api/iterator"
+
+	probepb "github.com/googleinterns/step224-2020/config/proto"
+	m "github.com/googleinterns/step224-2020/hermes/probe/metrics"
+)
+
+// deleteRandomFile deletes a random file in the target storage system bucket.
+// It then checks that the file has been deleted by trying to get the object.
+// Arguments:
+// TODO(evanSpendlove): Update comment here
+//	- ctx: pass the context so this probe can be cancelled if needed.
+//	- config: pass the HermesProbeDef config for the probe calling this function.
+//	- target: pass the target run information.
+//	- client: pass an initialised storage client for this target system.
+//	- logger: pass the logger associated with the probe calling this function.
+// Returns:
+//	- fileID: returns the ID of the file delete OR a missing file to be created if one is found.
+//	- err:
+//		- [...].hermes_file_missing: the file to be deleted does not exist in Hermes' StateJournal.
```

Comment:
The reason that I titled this as "hermes_file_missing" is because the file is not found in Hermes' state, but it could exist on the storage system if the Journal is inconsistent due to some error/bug when updating the Journal.